src/custom-blocks/auto-item-list/index.js

```diff
@@ -0,0 +1,468 @@
+    },
+    listLength: {
+      type: "numeric",
+      default: 3
+    },
+    listExpiry: {
+      type: "numeric",
+      default: 0
+    },
+    listEmptyText: {
+      type: "string",
+      default: "No items to display."
+    },
+    listItemType: {
+      type: "string",
+      default: "post"
+    },
+    listSort: {
+      type: "string",
+      default: "date"
+    },
+    listDatePrefix: {
+      type: "string",
+      default: "Updated"
+    },
+    listImage: {
+      type: "boolean",
+      default: false
+    },
+    listDefaultImage: {
+      type: "string",
+      default: ""
+    },
+    pastFuture: {
+      type: "string",
+      default: "past"
+    }
+
+  },
+
+  edit: props => {
+    const {
+      setAttributes,
+      attributes: {
+        listLength,
+        listExpiry,
+        listEmptyText,
+        listHasDate,
+        listHasTime,
+        listItemType,
+        listSort,
+        listDatePrefix,
+        listImage,
+        listDefaultImage,
+        pastFuture
+      },
+      className
+    } = props
+
+    const {
+      allPostTypes,
+    } = useSelect(
+      ( select ) => {
+          const { getPostTypes } = select(
+          coreStore
+        );
+        const allPostTypesList = getPostTypes(
+          { per_page: -1 }
+        );
+        return {
+          allPostTypes: allPostTypesList
+        };
+      }
+    );
+
+    let itemTypes = [{
+      label: "-",
+      value: ""
+    }]
+    let itemTypesFinishedParsing = false;
+
+    let fieldOptions = [
+      {label: "Publish date", value: "date"},
+      {label: "Modified date", value: "modified"},
+    ]
+
+    if (allPostTypes) {
+      allPostTypes.forEach(thisPostType => {
+        if (thisPostType.name != "Media" && thisPostType.name != "Posts" && thisPostType.name != "Pages" && thisPostType.viewable) {
+          itemTypes.push({
+            label: thisPostType.name,
+            value: thisPostType.slug
+          });
+        }
+      });
+      itemTypesFinishedParsing = true;
+    }
+
+    const {
+      allDocuments,
+    } = useSelect(
+      ( select ) => {
+        if (listItemType) {
+
+          const { getEntityRecords } = select(
+            coreStore
+          );
+
+          const posts = getEntityRecords(
+            'postType',
+            listItemType,
+            { per_page: -1 }
+          );
+
+          return {
+            allDocuments: posts
+          };
+        } else {
+          return {
+            allDocuments: false
+          };
+        }
+      }
+    );
+
+    if (Array.isArray( allDocuments ) && allDocuments.length >= 1) {
+      // This section loops through the properties of this custom post type (first one, they'll all be the same)
+      // We are looking for any other fields which are dates
+      if (allDocuments[0].acf) {
+        for (var prop in allDocuments[0].acf) {
+          fieldOptions.push({label: (prop.charAt(0).toUpperCase()+prop.slice(1)).replaceAll("_"," "), value: prop});
```

Comment:
How does this know if it is a date? at the moment it looks as if it pulls all meta fields?

Reply:
my recommendation would be we remove this for now and add a cleaner solution after MVP